test/Autofac.Specification.Test/Features/CompositeTests.cs

```diff
@@ -0,0 +1,507 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Autofac.Core;
+using Autofac.Features.Metadata;
+using Autofac.Features.OwnedInstances;
+using Xunit;
+
+namespace Autofac.Specification.Test.Features
+{
+    public class CompositeTests
```

Comment:
Probably should test what happens if:
- Two composites are registered for the same interface.
- Composite is registered in the root container, then another composite (for the same interface) is registered in a child lifetime scope.
- Disposing the `Owned<Composite>` should also dispose the items in the composite that aren't singletons (that is, ensuring that the items in the `Owned<Composite>` share the lifetime scope that wraps that `Owned<Composite>`).
- Register a composite with an `InstancePerMatchingLifetimeScope(x)` and then resolve the interface type in a lifetime scope that _doesn't_ match.
- Can a registration source provide composites? (Maybe that's just a question I have and not a test?)

Reply:
Tests added; sources can add composites, even if it's not super elegant right now, so I've tested that.

Could you take a look at the below tests to make sure the expectations are correct?

- `CompositeForNamedScopeThrowsExceptionIfServiceResolvedInDifferentScope`
- `CompositeForNamedScopeResolvedInNestedScope`
- `ManualCompositeOverridesSourcedComposite`
- `SourceInNestedScopeOverridesManualComposite`

Reply:
I think the other tests look great.